swift/ql/test/library-tests/dataflow/flowsources/alamofire.swift

```diff
@@ -0,0 +1,551 @@
+        completionHandler: @escaping (AFDownloadResponse<String>) -> Void) -> Self {
+            return self
+    }
+
+    @discardableResult
+    func responseJSON(
+        a1 : Int = 0, // these parameters do not matter for our purposes
+        a2 : Int = 0,
+        completionHandler: @escaping (AFDownloadResponse<Any>) -> Void) -> Self {
+            return self
+    }
+
+    @discardableResult
+    func responseDecodable<T: Decodable>(
+        of type: T.Type = T.self,
+        a1 : Int = 0, // these parameters do not matter for our purposes
+        a2 : Int = 0,
+        completionHandler: @escaping (AFDownloadResponse<T>) -> Void) -> Self {
+            return self
+    }
+}
+
+extension DataStreamRequest
+{
+    @discardableResult
+    func responseStream(
+        a1 : Int = 0, // these parameters do not matter for our purposes
+        a2 : Int = 0,
+        stream: @escaping Handler<Data, Never>) -> Self {
+            return self
+    }
+
+    @discardableResult
+    func responseStream<Serializer: DataStreamSerializer>(
+        using serializer: Serializer,
+        a1 : Int = 0, // these parameters do not matter for our purposes
+        a2 : Int = 0,
+        stream: @escaping Handler<Serializer.SerializedObject, AFError>) -> Self {
+            return self
+    }
+
+    @discardableResult
+    func responseStreamString(
+        a1 : Int = 0, // these parameters do not matter for our purposes
+        a2 : Int = 0,
+        stream: @escaping Handler<String, Never>) -> Self {
+            return self
+    }
+
+    @discardableResult
+    func responseStreamDecodable<T: Decodable>(
+        of type: T.Type = T.self,
+        a1 : Int = 0, // these parameters do not matter for our purposes
+        a2 : Int = 0,
+        stream: @escaping Handler<T, AFError>) -> Self {
+            return self
+    }
+}
+
+class Session {
+	static let `default` = Session()
+
+	func request(
+		_ convertible: URLConvertible,
+        a2 : Int = 0, // these parameters vary and do not matter for our purposes
+        a3 : Int = 0) -> DataRequest {
+			return DataRequest()
+	}
+
+	func download(
+		_ convertible: URLConvertible,
+        a2 : Int = 0, // these parameters vary and do not matter for our purposes
+        a3 : Int = 0,
+        to destination: DownloadRequest.Destination? = nil) -> DownloadRequest {
+			return DownloadRequest()
+	}
+
+	func streamRequest(
+		_ convertible: URLConvertible,
+        a2 : Int = 0, // these parameters vary and do not matter for our purposes
+        a3 : Int = 0) -> DataStreamRequest {
+			return DataStreamRequest()
+	}
+}
+
+let AF = Session.default
+
+// --- tests ---
+
+struct MySerializer: ResponseSerializer {
+    func serialize() -> String { return "" }
+    func serializeDownload() -> String { return "" }
+}
+
+struct MyStreamSerializer: DataStreamSerializer {
+    func serialize(_ data: Data) throws -> String { return "" }
+}
+
+struct MyDecodable: Decodable {
+}
+
+func doSomething(_ p: Any) {}
```

Comment:
The syntax highlighting is awful from here and onwards in this file. Did a non-printable character sneak in here?

Reply:
Yeah, I noticed that, couldn't figure out what's causing it.  There don't seem to be any hidden characters (I tried deleting and re-typing several lines including this one).  My best guess is its a bug in the syntax highlighting???

Reply:
On another PR I saw red highlighting due to inconsistent indentation.  That doesn't seem to be exactly what's going on here, but a clue perhaps.

Reply:
I've adjusted this bit to try and make the red highlighting go away.  There's nothing telling us what the problem is, it could be a highlighting but, so I'm not inclined to take this too seriously if it persists.

Reply:
(this issue was reported to the appropriate team)